BalerAIDriver.lua

```diff
@@ -40,9 +40,8 @@ function BalerAIDriver:allFillLevelsOk()
 end
 
 function BalerAIDriver:handleBaler()
-	-- turn.lua will raise/lower as needed, don't touch the balers while the turn maneuver is executed
-	if self.turnIsDriving then return end
-
+	-- turn.lua will raise/lower as needed, don't touch the balers while the turn maneuver is executed or while on temporary alignment / connecting track
+	if self.turnIsDriving or self.fieldworkState == self.states.ON_CONNECTING_TRACK or self.fieldworkState == self.states.TEMPORARY then return end
```

Comment:
Great fix. Would you mind moving this condition to a function something like isOkToDropBale() so we have it in one place?

Reply:
No problem. Here you go: https://github.com/Courseplay/courseplay/pull/3722/commits/76999eb0d108486671bf96f052560994735614f8

------------------------

One further annotation / question:

While testing, I found that no-combo round balers would still drop their bales on alignment / temporary courses if starting with a full baler (rare occassion, e.g. if someone baled manually and forgot to empty the baler before using courseplay) . Adding **`self.baler:setIsUnloadingBale(false)`**  on the BalerAIDriver's **`if not self:handlingIsAllowed()...`** statement kinda does the job (bales actually don't get dropped), but the unload animation itself still plays.

Do you know which code else than BalerAIDriver may be responsible for emptying the baler / triggering the unload animation? I couldn't find any other code responsible for that...TY beforehand.

Reply:
Ok, but why do we need two functions? Why can't we use the BalerAIDriver:handlingIsAllowed() (isHandlingAllowed() may be better) instead of BaleWrapper:isOkToDropBale()?

Reply:
You are right...there's actually no need for two separate functions, as they are identical and do the same. 

Sometimes, I'm a little bit anxious about future-proofing code (makes some of my coding too complex). Also, I wanted to respect your class-type-approach (as I understand, each type - baler, wrapper, baler, combo etc - having it's own lua) by trying to keep both luas independet from each other as much as possible.

On afterthought, I may have overcomplicated my coding a little bit. Will simplify as you requested.

Reply:
Done: https://github.com/Courseplay/courseplay/pull/3722/commits/505982116dae827d582a470648f069dac0dece41